File: intermediate/javascript/pyramids.js
function pyramids(n) {
  const colCount = 2 * n - 1;
  const midpoint = Math.floor(colCount / 2);
  for (let row = 0; row < n; row++) {
    let level = '';
    for (let col = 0; col < colCount; col++) {
      let range = (midpoint - row >= col) &&  (midpoint + row <= col);
      if (range) {
        level += '#';
      } else {
        level += ' ';
      }
    }
    console.log(level)
  }
}

pyramids(9)

File: intermediate/python/pyramids.py
from math import floor

def pyramid(n):
  col_count = n * 2 - 1
  midpoint = floor(col_count /2)
  for row in range(n):
    level = ''
    for col in range(col_count):
      pyramid_range = (midpoint - row <= col) and (midpoint + row >= col)
      if pyramid_range:
        level += '#'
      else:
        level += ' '
    print(level)

def pyramidRecursive(n, row = 0, col = 0, level = ''):
  colCount = n * 2 - 1
  midpoint = floor(colCount / 2)
  pyramid_range = (midpoint - row <= col) and (midpoint + row >= col)

  if row == n: return

  if len(level) == colCount:
    print(level)
    return pyramidRecursive(n, row + 1)

  if pyramid_range:
    level += '#'
  else: level += ' '

  return pyramidRecursive(n, row, col + 1, level)

pyramidRecursive(9)